test/integration/segmentation_camera_plugin.cc

```diff
@@ -0,0 +1,305 @@
+
+  memcpy(g_buffer, _msg.data().c_str(), size);
+  g_counter++;
+  g_mutex.unlock();
+}
+
+/// \brief wait till you read the published frame
+void WaitForNewFrame()
+{
+  auto waitTime = std::chrono::duration_cast< std::chrono::milliseconds >(
+  std::chrono::duration< double >(0.001));
+  int counter = 0;
+  // wait for the counter to increase
+  for (int sleep = 0; sleep < 300 && counter == 0; ++sleep)
+  {
+    g_mutex.lock();
+    counter = g_counter;
+    g_mutex.unlock();
+    std::this_thread::sleep_for(waitTime);
+  }
+  EXPECT_GT(counter, 0);
+}
+
+/// \brief Build the scene with 3 boxes besides each other
+/// the 2 aside boxes have the same label & the middle is different
+void BuildScene(rendering::ScenePtr scene)
+{
+  math::Vector3d leftPosition(3, -1.5, 0);
+  math::Vector3d rightPosition(3, 1.5, 0);
+  math::Vector3d middlePosition(3, 0, 0);
+
+  rendering::VisualPtr root = scene->RootVisual();
+
+  // create box visual
+  rendering::VisualPtr box = scene->CreateVisual();
+  box->AddGeometry(scene->CreateBox());
+  box->SetOrigin(0.0, 0.0, 0.0);
+  box->SetLocalPosition(leftPosition);
+  box->SetLocalRotation(0, 0, 0);
+  box->SetUserData("label", 1);
+  root->AddChild(box);
+
+  // create box visual of same label
+  rendering::VisualPtr box1 = scene->CreateVisual();
+  box1->AddGeometry(scene->CreateBox());
+  box1->SetOrigin(0.0, 0.0, 0.0);
+  box1->SetLocalPosition(rightPosition);
+  box1->SetLocalRotation(0, 0, 0);
+  box1->SetUserData("label", 1);
+  root->AddChild(box1);
+
+  // create box visual of different label
+  ignition::rendering::VisualPtr box2 = scene->CreateVisual();
+  box2->AddGeometry(scene->CreateBox());
+  box2->SetOrigin(0.0, 0.0, 0.0);
+  box2->SetLocalPosition(middlePosition);
+  box2->SetLocalRotation(0, 0, 0);
+  box2->SetUserData("label", 2);
+  root->AddChild(box2);
+}
+
+/////////////////////////////////////////////////
+void SegmentationCameraSensorTest::ImagesWithBuiltinSDF(
+  const std::string &_renderEngine)
+{
+  std::string path = ignition::common::joinPaths(PROJECT_SOURCE_PATH, "test",
+      "integration", "segmentation_camera_sensor_builtin.sdf");
+  sdf::SDFPtr doc(new sdf::SDF());
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  int width = imagePtr->Get<int>("width");
+  int height = imagePtr->Get<int>("height");
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
```

Comment:
Before getting the engine, we should also make sure that `_renderEngine` is `ogre2` since the segmentation camera has only been implemented for `ogre2`. Here's an example from the thermal camera (in the thermal camera, `ogre` and `ogre2` are supported, but in our case, we just want `ogre2`): https://github.com/ignitionrobotics/ign-sensors/blob/a9517f8b7788838d373143fe87582ca02a8d7c89/test/integration/thermal_camera_plugin.cc#L137-L144